src/Microsoft.FeatureManagement/ConfigurationFeatureSettingsProvider.cs

```diff
@@ -39,17 +40,53 @@ public void Dispose()
             _changeSubscription = null;
         }
 
-        public IFeatureSettings TryGetFeatureSettings(string featureName)
+        public Task<FeatureSettings> GetFeatureSettingsAsync(string featureName)
         {
+            if (featureName == null)
+            {
+                throw new ArgumentNullException(nameof(featureName));
+            }
+
             if (Interlocked.Exchange(ref _stale, 0) != 0)
             {
                 _settings.Clear();
             }
 
-            return _settings.GetOrAdd(featureName, (name) => ReadFeatureSettings(name));
+            //
+            // Query by feature name
+            FeatureSettings settings = _settings.GetOrAdd(featureName, (name) => ReadFeatureSettings(name));
+
+            return Task.FromResult(settings);
         }
 
-        private IFeatureSettings ReadFeatureSettings(string featureName)
+        //
+        // The async key word is necessary for creating IAsyncEnumerable.
+        // The need to disable this warning occurs when implementaing async stream synchronously. 
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        public async IAsyncEnumerable<FeatureSettings> GetAllFeatureSettingsAsync()
+#pragma warning restore CS1998
+        {
+            if (Interlocked.Exchange(ref _stale, 0) != 0)
+            {
+                _settings.Clear();
+            }
+
+            var featureSettings = new List<FeatureSettings>();
+
+            //
+            // Query all
+            foreach (string featureName in GetFeatureConfigurationSections().Select(s => s.Key))
```

Comment:
Will it be possible for `GetFeatureConfigurationSections()` to return `null`?

Reply:
No, that's a private method where we don't return null.